hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/CallerContext.java

```diff
@@ -121,6 +121,44 @@ public String toString() {
     return str;
   }
 
+  private static String parseSpecialValue(String content, String key) {
```

Comment:
This method is duplicate with `NameNode#parseSpecialValue`, can you remove the method in `NameNode.class` and use this one?

Reply:
I agree with this suggestion.

Reply:
CLIENT_ID_STR and CLIENT_CALL_ID_STR are involved here, so I keep them for now.
It might be more appropriate to create a new jira to solve this problem?
What do you think, @ZanderXu .

Reply:
Sorry, I didn't  get you idea. Just public `parseSpecialValue` method in `CallerContext.java` and remove `parseSpecialValue` method in `NameNode.class`, and modify the caller place in `NameNode.java` to use `CallerContext. parseSpecialValue()`.